### PR TITLE
Add Versioned AWS Lambda orb

### PR DESCRIPTION
**Pull request by tom-g-dane**
Add an orb for the migration team AWS lambda builds.

Ran `circleci orb validate migration-lambdas/orb.yml` and `circleci orb process migration-lambdas/orb.yml` locally and all looks good.

### CODE REVIEW

**Comment by tom-g-dane** [comment]
I did some mild formatting in some of the common files (IDE did some of it) let me know if you want me to revert this.

**Comment by tom-g-dane** [comment]
Marked as draft for now as feedback from Dave Connell is to make it more generic as it may be useful for teams outside of orion migrations.

**Comment by tom-g-dane** [comment]
Updated the orb to be non-migration specific

File: versioned-lambda/orb.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+            zip -r -q $ZIPFILE dist node_modules
+            aws s3 cp $ZIPFILE $S3PATH/$ZIPFILE
+
+  create-lambda-version:
+    parameters:
+      build-bucket:
+        description: "AWS S3 bucket containing lambda build artifacts"
+        type: string
+        default: ovo-orion-versioned-build
```
Comment (whyman) [inline versioned-lambda/orb.yml:82]: Should this be a default considering possible public usage?

Comment (tom-g-dane) [reply versioned-lambda/orb.yml:82]: wooops, yeah that shouldn't be a default. Will fix now thanks!

**Pull request merged**